<?xml version="1.0"?>
<robot name="my_lidar_bot">

  <link name="base_link"/>

  <link name="lidar_tilt_link"/>

  <joint name="servo_mount_joint" type="revolute">
    <parent link="base_link"/>
    <child link="lidar_tilt_link"/>
    <origin xyz="0 0 0.47" rpy="0 0 0"/> 
    <axis xyz="0 1 0"/> <limit lower="-1.57" upper="1.57" effort="10.0" velocity="1.0"/>
  </joint>

  <link name="laser_frame">
    <visual>
      <geometry>
        <cylinder length="0.04" radius="0.035"/>
      </geometry>
      <material name="red">
        <color rgba="0.8 0 0 1"/>
      </material>
    </visual>
  </link>

  <joint name="lidar_mount_joint" type="fixed">
    <parent link="lidar_tilt_link"/>
    <child link="laser_frame"/>
    <origin xyz="0.005 0 0.07" rpy="3.14159 0 0"/>
  </joint>

</robot>